### PR TITLE
[1LP][RFR] Gap Collection C&U

### PR DESCRIPTION
**Pull request by digitronik**
Purpose or Intent
=================
- Gap collection tests for VM and Host

{{pytest: cfme/tests/candu/test_gap_collection.py -v --use-provider complete}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit cf911b6b6c4487eb9104da9a91ee39d76929de22:

`widgetastic_manageiq/__init__.py`:
- :red_circle: Line 6:1: **F401** *'datetime.datetime' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/candu/test_gap_collection.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+
+@pytest.fixture(scope='module')
+def order_data(appliance, provider, enable_candu):
+    # Order two day back gap collection data for testing
+    end_date = datetime.now()
+    start_date = end_date - timedelta(days=2)
+
+    view = navigate_to(appliance.server.zone, 'CANDUGapCollection')
+    view.candugapcollection.end_date.fill(end_date)
```
Comment (rlbabyuk) [inline cfme/tests/candu/test_gap_collection.py]: Required: 
view.candugapcollection.fill{
'end_date': end_date,
'start_date': start_date
}

File: cfme/tests/candu/test_gap_collection.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        * C&U enabled appliance
+
+    Steps:
+        * Navigate to Configuration > Diagnostics > Zone Gap Collection Page
+        * Order old data
+        * Navigate to VM or Host Utilization page
+        * Check for Hourly data
+        * Check for Daily data
+    """
+    if element == 'host':
```
Comment (rlbabyuk) [inline cfme/tests/candu/test_gap_collection.py]: Required: This looks like a setup, Suggest moving it to parametrize fixture which will return you a correct object
line 51-59

File: cfme/tests/candu/test_gap_collection.py (hunk 1 above)
Comment (digitronik) [reply cfme/tests/candu/test_gap_collection.py]: Yes, Its possibly right solution but I am adding it as workaround means first fill `end_date` and after that `start_date`. As per my knowledge; passing `dict` not fallow order. Please, correct me If I am wrong. As I am playing with `all Vsphere providers`. I need to execute it for each and `calendar` will block selection, if I select first `start_date` then end date will `disable` as gap collection already executed for `first provider`.

Comment (rlbabyuk) [reply cfme/tests/candu/test_gap_collection.py]: They are filled in the same order as listed in the view. In your view, start_date is the first, and it filled first. If end date always should be filled fist just move it to the top in view class

File: cfme/tests/candu/test_gap_collection.py (hunk 2 above)
Comment (digitronik) [reply cfme/tests/candu/test_gap_collection.py]: As per our discussion on IRC, fixture parameter will run before provider. So, better to go with same. Its faster (for one provider run all `vm-hourly`, `vm-daily`, and `host-hourly`, `host-daily`) and nice id.

**Comment by digitronik** [comment]
@rlbabyuk: I added changes

**Pull request merged**